### ISSUE TITLE
Automatically use oversubscribtion flags when configuring with MPI 

### ISSUE THREAD

**Issue by fsimonis** [precice/precice#731]
**Please describe the problem you are trying to solve.**
Configuring preCICE with `MPICommunication=ON` enables the execution of tests.
Some MPI implementations do not allow to oversubscribe the node (i.e. ask for more cores than available). As we have to run the tests on 4 ranks, this is problematic for single and dual-core systems.
We currently require the user to [pass required flags to cmake](https://github.com/precice/precice/wiki/Tests#testing-a-build).

**Describe the solution you propose.**
_Do nothing if the user provides custom flags_
Use CMake to [detect the processor count](https://cmake.org/cmake/help/v3.10/module/ProcessorCount.html).
When this count is less than 4 do:
1. Run a dummy program, e. g. `cmake --version` on 1 core to check that we can actually run programs with MPI
2. Run the dummy programm on 4 cores and check if oversubscribtion is allowed
3. If it fails: retry with known flags such as `--oversubscribe` and pick the first one where the program succeeds.
   Warn if no flag works.

**Describe alternatives you've considered**
Do nothing and rely on good documentation.

**Comment by MakisH** [comment]
I think we can improve a lot our default options and "good defaults is better than extensive documentation". I really like oversubscribing by default and just displaying a warning.

**Comment by fsimonis** [comment]
An alternative could be to always set the following flag when executing the tests:
`OMPI_MCA_rmaps_base_oversubscribe=1`.
This would be an OpenMPI specific solution for an OpenMPI specific problem.

**Comment by MakisH** [comment]
> An alternative could be to always set the following flag when executing the tests:
> `OMPI_MCA_rmaps_base_oversubscribe=1`.
> This would be an OpenMPI specific solution for an OpenMPI specific problem.

This works with OpenMPI 4.0.3 (Ubuntu 20.04).
```bash
cmake [normal options] ..
make
OMPI_MCA_rmaps_base_oversubscribe=1make test
```

I like this solution as it only sets an environment variable that OpenMPI recognizes. In any case, we should only modify the environment of the tests, not the complete user environment (not by `export` e.g.).

I understand that this also makes it much easier to implement for different cases: we can by default set this variable if we find OpenMPI and we don't need to filter per version etc. If OpenMPI picks it up, it is fine. If it does not, then it probably does not need it.

 We should test that it also works with OpenMPI 2 (Ubuntu 18.04) and 3.

**Comment by fsimonis** [comment]
Solved by #780
